docs/function.md

```diff
@@ -110,231 +108,144 @@ let eleven = addFive(6);
-  switch (radius) {
-  | None => startAt(1, 1)
-  | Some(r_) => startAt(r_, r_)
-  }
-};
+/* Position (10, 0) with radius 2 */
+makeCircle(~x=10, ~radius=2, ());
 ```
 
-When given in this syntax, `radius` is **wrapped** in the standard library's `option` type, defaulting to `None`. If provided, it'll be wrapped with a `Some`. So `radius`'s type value is either `None` or `Some(int)` here.
+Notice that an extra unit argument was added after the `radius` argument. This
+has to do with the partial application feature. Now that some arguments have
+default values and can be left out the compiler needs some indication of when
+you are "done" calling a function.
 
-**Note**: `None | Some(foo)` is a data structure type called variant, described [earlier](variant.md). This particular variant type is provided by the standard library. It's called `option`. Its definition: `type option('a) = None | Some('a)`.
+There must be a positional argument after all optional arguments so the compiler
+knows whether to partially apply a function or to use the default values.
 
-**Note** the unit `()` at the end of `drawCircle`. Writing this particular function without the unit `()` would lead to the following problem. Because `radius` and `color` are both labeled, the function can be curried, and it can be applied out-of-order, it's unclear what the following means:
+Consider if there were no final unit argument, what should this do:
 
 ```reason
-let whatIsThis = drawCircle(~color);
+makeCircle(~radius=1);
 ```
 
-Is `whatIsThis` a curried `drawCircle` function, waiting for the optional `radius` to be applied? Or did it finish applying because the `radius` is optional? To address this confusion, append a positional (aka non-labeled) argument to `drawCircle` (conventionally `()`), and OCaml will, as a rule of thumb, presume the optional labeled argument is omitted when the positional argument is provided.
+It is not clear if that should return a function waiting for `x` and `y`
+coordinates, or if should default the `x` and `y` coordinates to zeroes. With
+the final unit argument these cases are disambiguated:
 
-If we don't supply the unit, OCaml knows we want to curry the function.
 ```reason
-let curriedFunction = drawCircle(~color);
-```
+/* Function waiting for x and y coordinates. */
+let makeUnitCircle = makeCircle(~radius=1);
+let c = makeUnitCircle(~x=5, ~y=5, ());
 
-If we _do_ supply the unit, OCaml knows we deliberately omit the `radius` parameter, and the function is executed.
-```reason
-let circle = drawCircle(~color, ());
+/* Creates a circle at (0, 0) with radius 1 */
+let c = makeCircle(~radius=1, ());
 ```
 
-### Explicitly Passed Optional
+### Optional Arguments
 
-Sometimes, you might want to forward a value to a function without knowing whether the value is `None` or `Some(a)`. Naively, you'd do:
+Similar to default argument values, named arguments can be marked as optional.
+This is just syntax around making the default value `None`.
```

Comment:
This is not exactly correct:

https://sketch.sh/s/yJfd9kE6DmzSgxP4dx9LBJ/

- A question optional means "there is no default for this argument, and inside of the function, the variable will be either `None` (if no argument was passed) or `Some(arg)` if argument `arg` was passed.
- A default value of `~arg=None`, means that the argument supplied must be an optional, and if no `~arg` argument is supplied, the `arg` identifier inside the function will receive the `None` value by default. If an argument `~arg` _was_ supplied then the value of `arg` inside of the function will be the value passed to the function when called - which will be an optional value.

Reply:
I think my comment above might provide an alternative explanation that clarifies this.